plugin/storage/clickhouse/factory.go

```diff
@@ -0,0 +1,71 @@
+// Copyright (c) 2023 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package clickhouse
+
+import (
+	"context"
+	"database/sql"
+
+	"go.opentelemetry.io/collector/pdata/ptrace"
+	"go.uber.org/zap"
+
+	"github.com/jaegertracing/jaeger/pkg/metrics"
+	chSpanStore "github.com/jaegertracing/jaeger/plugin/storage/clickhouse/spanstore"
+	"github.com/jaegertracing/jaeger/storage/dependencystore"
+	"github.com/jaegertracing/jaeger/storage/spanstore"
+)
+
+type Factory struct {
+	client         *sql.DB
+	spansTableName string
+	logger         *zap.Logger
+}
+
+func NewFactory(ctx context.Context, cfg Config, logger *zap.Logger) *Factory {
+	f := &Factory{
+		logger:         logger,
+		spansTableName: cfg.SpansTableName,
+	}
+
+	client, err := cfg.NewClient(ctx)
+	if err != nil {
+		f.logger.Error("failed to create ClickHouse client", zap.Error(err))
+	} else {
+		f.client = client
+	}
+
+	return f
+
+	//TODO: Move some steps to Initialize()
+}
+
+func (f *Factory) CreateSpansTable(ctx context.Context) {
```

Comment:
why is this function public, who would call it?

Reply:
Currently, Jaeger V2's exporter checks the type of Factory. If it's a clickhouse Factory, then f.CreateSpansTable() is called, as you can see in this [piece](https://github.com/jaegertracing/jaeger/pull/4941/files#diff-d11f17877ed77e35143c698010924340c6cf95489656e3ccb0b3c2654c9b9853R42-R52).